rust/arrow/src/ffi.rs

```diff
@@ -193,6 +216,42 @@ fn to_datatype(format: &str) -> Result<DataType> {
         "ttm" => DataType::Time32(TimeUnit::Millisecond),
         "ttu" => DataType::Time64(TimeUnit::Microsecond),
         "ttn" => DataType::Time64(TimeUnit::Nanosecond),
+
+        // Note: The datatype null will only be created when called from ArrowArray::buffer_len
+        // at that point the child data is not yet known, but it is also not required to determine
+        // the buffer length of the list arrays.
+        "+l" => {
+            let nullable = schema.flags == 2;
+            // Safety
+            // Should be set as this is expected from the C FFI definition
+            debug_assert!(!schema.name.is_null());
+            let name = unsafe { CString::from_raw(schema.name as *mut c_char) }
```

Comment:
This doesn't seem right. The [doc](https://doc.rust-lang.org/std/ffi/struct.CString.html#method.from_raw) for `from_raw` says: 

> This should only ever be called with a pointer that was earlier obtained by calling CString::into_raw. Other usage (e.g., trying to take ownership of a string that was allocated by foreign code) is likely to lead to undefined behavior or allocator corruption.

But we are exactly in the case where `schema.name` can have been allocated by C++ or Python or anything else. It seems instead you should use [CStr](https://doc.rust-lang.org/std/ffi/struct.CStr.html) instead: "Representation of a borrowed C string".